Components/Motors/VESC.cs

```diff
@@ -0,0 +1,190 @@
+﻿using System;
+using Scarlet.Filters;
+using Scarlet.IO;
+using Scarlet.Utilities;
+using System.Threading;
+using System.Collections.Generic;
+
+namespace Scarlet.Components.Motors
+{
+    public class VESC : IMotor
+    {
+        private IFilter<float> Filter; // Filter for speed output
+        private readonly IUARTBus UARTBus;
+        private readonly int CANForwardId;

```

Comment:
As per the style guidelines, abbreviations like ID should remain properly capitalized, so this should be `CANForwardID`.